src/pages/index.js

```diff
@@ -72,50 +88,166 @@ newCardPopup.setEventListeners();
 const popupImage = new PopupWithImage("#preview__modal");
 popupImage.setEventListeners();
 
+const avatarPopup = new PopupWithForm("#avatar__modal", handleAvatarUpdate);
+avatarPopup.setEventListeners();
+
+const deleteConfirm = new PopupWithConfirm("#delete__modal", handleCardDelete);
+deleteConfirm.setEventListeners();
+
+const api = new Api({
+  baseURL: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "4792ec92-cf1c-45a6-8740-0f5d63585faa",
+    "Content-type": "application/json",
+  },
+});
+
 //////////////////////////////////////////// FUNCTIONS //////////////////////////////////////////////////////
 
 function handleImageClick(name, link) {
   popupImage.openPopup({ name, link });
 }
 
 function handleProfileModalSubmit(profileData) {
+  profileModalSubmitButton.textContent = "Saving...";
   console.log("profileData:", profileData);
   const name = profileData.title; //set name var to work with userInfo class
   const job = profileData.description; //set job var to work with userInfo class
   console.log("name:", name, "job:", job); //debug log
   user.setUserInfo({ name, job }); //set user information
+  api.updateUserInfo(profileData.title, profileData.description);
   const updatedUserInfo = user.getUserInfo(); //get updated information and store in a var
   console.log(updatedUserInfo); // debug log
   profilePopup.closePopup(); //close popup
```

Comment:
* You need to close modals only in blocks `then` after a **successful response** from the server so that when there is an error, it would be opened (the inputs will not be cleared and the user could send the data again to the server), and if you change the button text, you could see the change as well

Reply:
You haven't corrected my remark https://github.com/RealnNTrue16/se_project_aroundtheus/pull/3#discussion_r1697656722